70_ima_policy.md

```diff
@@ -0,0 +1,348 @@
+  appropriate SIG(s).
+- [ ] **Merge early and iterate.**
+  Avoid getting hung up on specific details and instead aim to get the goals of
+  the enhancement clarified and merged quickly.  The best way to do this is to just
+  start with the high-level sections and fill out details incrementally in
+  subsequent PRs.
+-->
+# enhancement-70: Merge allow and exclude list into an IMA policy
+
+<!--
+This is the title of your enhancement.  Keep it short, simple, and descriptive.  A good
+title can help communicate what the enhancement is and should be considered as part of
+any review.
+-->
+
+<!--
+A table of contents is helpful for quickly jumping to sections of a enhancement and for
+highlighting any additional information provided beyond the standard enhancement
+template.
+-->
+
+<!-- toc -->
+- [Release Signoff Checklist](#release-signoff-checklist)
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories (optional)](#user-stories-optional)
+    - [Story 1](#story-1)
+    - [Story 2](#story-2)
+  - [Notes/Constraints/Caveats (optional)](#notesconstraintscaveats-optional)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Test Plan](#test-plan)
+  - [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (optional)](#infrastructure-needed-optional)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+<!--
+**ACTION REQUIRED:** In order to merge code into a release, there must be an
+issue in [keylime/enhancements] referencing this enhancement and targeting a release**.
+
+For enhancements that make changes to code or processes/procedures in core
+Keylime i.e., [keylime/keylime], we require the following Release
+Signoff checklist to be completed.
+
+Check these off as they are completed for the Release Team to track. These
+checklist items _must_ be updated for the enhancement to be released.
+-->
+
+- [ ] Enhancement issue in release milestone, which links to pull request in [keylime/enhancements]
+- [ ] Core members have approved the issue with the label `implementable`
+- [ ] Design details are appropriately documented
+- [ ] Test plan is in place
+- [ ] User-facing documentation has been created in [keylime/keylime-docs]
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+## Summary
+
+This enhancement proposal changes and unifies the IMA configuration options by
+formally specifying a JSON schema. The tenant and verifier are modified to use
+the IMA policy instead of the old flat file format to simplify the process and
+allow further development like server side signature validation.
+
+For users of the old flat file format a upgrade path using a commandline tool
+will be provided.
+
```

Comment:
Kinda, but not really. Ideally we want to make the tenant simpler to use and not more complicated. Removing and providing a tool is easier than trying to build backwards compatability into the tenant long term.

When I asked there was still the need for the flat file format (I think it came from @maugustosilva) and with that approach we can support as long as needed, but no longer as a main part of the tenant.

Reply:
My main concern is that `keylime_tenant` for most people is THE interface for keylime and making backwards incompatible changes without a deprecation cycle is going to cause more and more problems as adoption increases. Especially after it starts landing in major OS distros soon. Maybe that's not an issue with this change, but I'm concerned about the precedent it sets.

Maybe a compromise is to have the tenant accept all of the old arguments and do one of the following:

1. Issue a deprecation warning and use the new tool in the background to do the conversion.
2. Issue an error message and helpfully point the user at the new conversion tool

In my mind the 1st is a superior user experience, but I can probably be convinced that the 2nd is better to keep the code cleaner.

Reply:
I think we should think about how many people are actually using the IMA feature today and make our decision based on that. I think that for new users the current tenant commandline interface is not really usable. It has way too many options, which are not really used. I'm fine with implementing the second idea.

When Keylime ships in e.g. RHEL it will be probably frozen on one mayor version right? So then it will be always an upgrade issue between the last shipped one and the current master.